extension/entitystore/factory.go

```diff
@@ -13,10 +14,16 @@ import (
 var (
 	TypeStr, _  = component.NewType("entitystore")
 	entityStore *EntityStore
+	mutex       sync.RWMutex
 )
 
 func GetEntityStore() *EntityStore {
-	return entityStore
+	mutex.RLock()
+	defer mutex.RUnlock()
+	if entityStore != nil && entityStore.ready {
```

Comment:
The lock only protects the `entityStore`. It doesn't protect multiple threads from reading/writing to the `ready` bool.

Reply:
We are writing ready bool only once in entitystore extension Start method. Do we still need to have lock on it?